### PR TITLE
gurk-rs: 0.5.1 -> 0.5.2

### PR DESCRIPTION
**Pull request by kachick**
Changes: https://github.com/boxdot/gurk-rs/compare/v0.5.1...v0.5.2

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by kachick** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 356353`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>gurk-rs</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>gurk-rs</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>gurk-rs</li>
  </ul>
</details>

**Review by devhell — APPROVED** [review]
Thank you! :)

File: pkgs/applications/networking/instant-messengers/gurk-rs/default.nix [hunk 1]
```diff
@@ -11,29 +11,22 @@
 
 rustPlatform.buildRustPackage rec {
   pname = "gurk-rs";
-  version = "0.5.1";
+  version = "0.5.2";
 
   src = fetchFromGitHub {
     owner = "boxdot";
     repo = pname;
     rev = "v${version}";
```
Comment (khaneliman) [inline pkgs/applications/networking/instant-messengers/gurk-rs/default.nix:19]:
```suggestion
    repo = "gurk-rs";
    rev = "refs/tags/v${version}";
```

**Comment by khaneliman** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 356353`

---
### `x86_64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>gurk-rs</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>gurk-rs</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>gurk-rs</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>gurk-rs</li>
  </ul>
</details>

File: pkgs/applications/networking/instant-messengers/gurk-rs/default.nix [hunk 2]
```diff
@@ -11,29 +11,22 @@
...
-      "libsignal-protocol-0.1.0" = "sha256-4aHINlpVAqVTtm7npwXQRutZUmIxYgkhXhApg7jSM4M=";
-      "libsignal-service-0.1.0" = "sha256-AOGw76A9R2qH3hc7B+MBE3okzW8b5LTZdepzUDOv9lM=";
-      "curve25519-dalek-4.1.3" = "sha256-bPh7eEgcZnq9C3wmSnnYv0C4aAP+7pnwk9Io29GrI4A=";
-      "presage-0.6.2" = "sha256-t9t8ecPtefI/jYlk+Ul8WdgH26VJIkfMptbKxprekS0=";
-      "qr2term-0.3.1" = "sha256-U8YLouVZTtDwsvzZiO6YB4Pe75RXGkZXOxHCQcCOyT8=";
-    };
-  };
+  useFetchCargoVendor = true;
+
+  cargoHash = "sha256-YeRrd2ybCTVAu7EyAZg82eqUP9wvsA8GArdn0oY1eMg=";
```
Comment (khaneliman) [inline pkgs/applications/networking/instant-messengers/gurk-rs/default.nix:29]:
```suggestion
  cargoHash = "sha256-jTZ2wJPXj3nU7GVTfne64eSra+JuKhNryCtRZMKOE44=";
```
Getting hash mismatch from nixpkgs-reviews.

**Comment by khaneliman** [comment]
@ofborg build

**Review by khaneliman — COMMENTED** [review]
Hmm... seeing hash mismatch

File: pkgs/applications/networking/instant-messengers/gurk-rs/default.nix (hunk 2 above)
Comment (kachick) [reply pkgs/applications/networking/instant-messengers/gurk-rs/default.nix]: Hmm... after this change, my device makes following error

```console
> nix build .#gurk-rs                                                                                   
error: hash mismatch in fixed-output derivation '/nix/store/r0zm2wwh4q3l1rj9jaj8fcgfq0qg2kn8-gurk-rs-0.5.2-vendor-staging.drv':
         specified: sha256-jTZ2wJPXj3nU7GVTfne64eSra+JuKhNryCtRZMKOE44=
            got:    sha256-YeRrd2ybCTVAu7EyAZg82eqUP9wvsA8GArdn0oY1eMg=
error: 1 dependencies of derivation '/nix/store/g22vkfpbqhyhmksdn4af54j5a8k31cl9-gurk-rs-0.5.2-vendor.drv' failed to build
error: 1 dependencies of derivation '/nix/store/4jcn8nscxrfr6hblszm0zrjf1fmvkbs0-gurk-rs-0.5.2.drv' failed to build
```

Comment (khaneliman) [reply pkgs/applications/networking/instant-messengers/gurk-rs/default.nix]: > Hmm... after this change, my device makes following error
> 
> ```
> > nix build .#gurk-rs                                                                                   
> error: hash mismatch in fixed-output derivation '/nix/store/r0zm2wwh4q3l1rj9jaj8fcgfq0qg2kn8-gurk-rs-0.5.2-vendor-staging.drv':
>          specified: sha256-jTZ2wJPXj3nU7GVTfne64eSra+JuKhNryCtRZMKOE44=
>             got:    sha256-YeRrd2ybCTVAu7EyAZg82eqUP9wvsA8GArdn0oY1eMg=
> error: 1 dependencies of derivation '/nix/store/g22vkfpbqhyhmksdn4af54j5a8k31cl9-gurk-rs-0.5.2-vendor.drv' failed to build
> error: 1 dependencies of derivation '/nix/store/4jcn8nscxrfr6hblszm0zrjf1fmvkbs0-gurk-rs-0.5.2.drv' failed to build
> ```

Well, that's super weird... I was wondering if something changed upstream in past few days but didn't seem like it... but, even more confusing if you're seeing a different hash than on my end. I had 4 different devices build it, too.

**Comment by khaneliman** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 356353`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>gurk-rs</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>gurk-rs</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>gurk-rs</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>gurk-rs</li>
  </ul>
</details>

File: pkgs/applications/networking/instant-messengers/gurk-rs/default.nix (hunk 2 above)
Comment (kachick) [reply pkgs/applications/networking/instant-messengers/gurk-rs/default.nix]: Thanks for sharing. :pray:  I have adding rebased commit as https://github.com/NixOS/nixpkgs/commit/9e335902fb3efcfbd279680aabcfdaa2005da342, now it does not make error for now.

```console
> nix build .#gurk-rs
> ./result/bin/gurk --version
gurk 0.5.2-dev
```

**Comment by kachick** [comment]
I don't have `aarch64-darwin` device, so checking it in GitHub hosted `macos-15` runner. It returns success :thinking: 

### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>gurk-rs</li>
  </ul>
</details>

**Comment by khaneliman** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 356353`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>gurk-rs</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>gurk-rs</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>gurk-rs</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>gurk-rs</li>
  </ul>
</details>

**Comment by khaneliman** [comment]
> I don't have `aarch64-darwin` device, so checking it in GitHub hosted `macos-15` runner. It returns success 🤔
> ### `aarch64-darwin`
> ✅ 1 package built:

I ran it again, it looked like a transient error from the new fetcher.

**Review by khaneliman — APPROVED** [review]

**Comment by TomaSajt** [comment]
The last force push  brought in the commit that now also fetches submodules.

nixpkgs-review failed because it is automatically rebasing on master, which already had that change.

the force push: https://github.com/NixOS/nixpkgs/compare/df54d02219152c3bfd69f268686e04ad17158a2f..9e335902fb3efcfbd279680aabcfdaa2005da342#diff-77d6ed0c1e97de400495829ad6c81d6da127ff83ff09d6d14ec9f887f46d4c67

**Comment by kachick** [comment]
Thank you for sharing that. The diff clarified why changed the result.

I have initially created this PR just after merging https://github.com/NixOS/nixpkgs/pull/349360, then master branch did not include https://github.com/NixOS/nixpkgs/pull/356415.

Anyway this `useFetchCargoVendor` much helped me, thank you! :pray:

**Comment by kachick** [comment]
Thank you! :pray:

**Pull request merged**